src/main/java/com/alvarium/hash/Md5Provider.java

```diff
@@ -0,0 +1,24 @@
+package com.alvarium.hash;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import com.alvarium.utils.Encoder;
+
+class Md5Provider implements HashProvider{
+  private final MessageDigest md5;
+
+  public Md5Provider() throws HashTypeException{
```

Comment:
Make this (and the other provider constructors) `protected` so that classes outside of this package can't arbitrarily instantiate the classes directly.

Reply:
fixed the md5 provider constructor visibility in 77f26aa0c8377b51ccf22dacbc86372d9dc7ceb9.